tests/common.go

```diff
@@ -3739,6 +3745,82 @@ func CreateAzureBackupLocation(name string, uid string, cloudCred string, cloudC
 	return nil
 }
 
+// WaitForBackupLocationAddition waits for backup location to be added successfully
+// or till timeout is reached. API should poll every `timeBeforeRetry` duration
+func WaitForBackupLocationAddition(
+	ctx context1.Context,
+	backupLocationName,
+	UID,
+	orgID string,
+	timeout time.Duration,
+	timeBeforeRetry time.Duration,
+) error {
+	req := &api.BackupLocationInspectRequest{
+		Name:  backupLocationName,
+		Uid:   UID,
+		OrgId: orgID,
+	}
+	f := func() (interface{}, bool, error) {
+		inspectBlResp, err := Inst().Backup.InspectBackupLocation(ctx, req)
+		if err != nil {
+			return "", true, err
+		}
+		actual := inspectBlResp.GetBackupLocation().GetBackupLocationInfo().GetStatus().GetStatus()
+		if actual == api.BackupLocationInfo_StatusInfo_Valid {
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("backup location status for [%s] expected was [%s] but got [%s]", backupLocationName, api.BackupLocationInfo_StatusInfo_Valid, actual)
+	}
+	_, err := task.DoRetryWithTimeout(f, timeout, timeBeforeRetry)
+	if err != nil {
+		return fmt.Errorf("failed to wait for backup location addition. Error:[%v]", err)
+	}
+	return nil
+}
+
+// CreateNFSBackupLocation creates backup location for nfs
+func CreateNFSBackupLocation(name string, uid string, orgID string, encryptionKey string, validate bool) error {
```

Comment:
Do we need to consider NFS versions ?

Reply:
CreateNFSBackupLocation does not create an NFSBackupLocation itself. Instead, it adds a pre-provisioned NFS BackupLocation, whose details are fetched from the environment, to Px-Backup.